Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -91,8 +93,15 @@ void RewriteGlobalVariables::run()
 {
     collectGlobals();
     insertStructs();
+    PipelineLayout pipelineLayout;
     for (auto& entryPoint : m_callGraph.entrypoints())
-        visitEntryPoint(entryPoint.function);
+        visitEntryPoint(entryPoint.function, entryPoint.stage, pipelineLayout);
+
+    for (auto& entryPoint : m_callGraph.entrypoints()) {
+        auto it = m_result.entryPoints.find(entryPoint.function.name());
+        ASSERT(it != m_result.entryPoints.end());
+        it->value.defaultLayout = pipelineLayout;
```

Comment:
I'm confused. Doesn't this mean all entry points get the same pipeline layout? Isn't that wrong?

Reply:
It seems like the compiler should compute a layout for each entry point and then the pipeline should merge the layouts during pipeline creation